flink-kubernetes-operator/src/main/java/org/apache/flink/kubernetes/operator/observer/Observer.java

```diff
@@ -54,10 +56,19 @@ public Observer(FlinkService flinkService, FlinkOperatorConfiguration operatorCo
         this.operatorConfiguration = operatorConfiguration;
     }
 
-    public boolean observe(
-            FlinkDeployment flinkApp, Context context, Configuration effectiveConfig) {
+    public void observe(FlinkDeployment flinkApp, Context context, Configuration effectiveConfig) {
+
         observeJmDeployment(flinkApp, context, effectiveConfig);
-        return isReadyToReconcile(flinkApp, effectiveConfig);
+        JobManagerDeploymentStatus deploymentStatus =
+                flinkApp.getStatus().getJobManagerDeploymentStatus();
+
+        if (deploymentStatus == JobManagerDeploymentStatus.READY
+                && flinkApp.getSpec().getJob() != null) {
```

Comment:
We can consider introducing a JobObserver/SessionObserver pair too similar to JobReconciler/SessionReconciler pair.

Reply:
I've opened a Jira for this https://issues.apache.org/jira/browse/FLINK-26546

Reply:
Thanks for the ticket, I improved this check to be more descriptive